components/layout_2020/flow/inline.rs

```diff
@@ -1314,13 +1318,56 @@ impl TextRun {
             // This prevents whitespace from being added to the beginning of a line. We could
             // trim it later, but we don't want it to come into play when determining line
             // width.
-            if run.glyph_store.is_whitespace() &&
-                !white_space.preserve_spaces() &&
-                !ifc.current_line.has_content
-            {
+            let is_non_preserved_whitespace =
+                run.glyph_store.is_whitespace() && !white_space.preserve_spaces();
+            if is_non_preserved_whitespace && !ifc.current_line.has_content {
                 continue;
             }
 
+            let new_advance_from_glyph_run = Length::from(run.glyph_store.total_advance());
+            let new_total_advance = new_advance_from_glyph_run +
```

Comment:
I wouldn't cal this "advance" if it includes `ifc.current_line.inline_position`. Maybe `new_inline_position`, `inline_end_position` or something.

Reply:
I've done the following renames:
 - `new_total_advance` -> `new_potential_inline_end_position`
 - `advance_from_text_run` -> `text_run_inline_size`
 - `new_advance_from_glyph_run` -> `advance_from_glyph_run`